frb_codegen/src/generator/dart/ty_delegate.rs

```diff
@@ -102,14 +123,36 @@ impl TypeDartGeneratorTrait for TypeDelegateGenerator<'_> {
     }
 
     fn structs(&self) -> String {
-        if let IrTypeDelegate::PrimitiveEnum { ir, .. } = &self.ir {
-            super::TypeEnumRefGenerator {
+        match &self.ir {
+            IrTypeDelegate::PrimitiveEnum { ir, .. } => super::TypeEnumRefGenerator {
                 ir: ir.clone(),
                 context: self.context.clone(),
             }
-            .structs()
-        } else {
-            "".into()
+            .structs(),
+            IrTypeDelegate::GeneralArray { length, .. }
+            | IrTypeDelegate::PrimitiveArray { length, .. } => {
+                format!(
+                    "class {0} {{
+                        final {1} inner;
+
+                        {0}(this.inner): assert(inner.length == {length});
+
+                        {0}.Unchecked(this.inner);
+
+                        {2} operator [](int index) {{
+                            return inner[index];
+                        }}
+                        
+                        void operator []=(int index, {2} value) {{
+                            inner[index] = value;
+                        }}
+                }}",
```

Comment:
A lot of common methods here can be factored out, perhaps something like this:
```dart
class TArrayN extends NonGrowableListView<T> {
  TArrayN(List<T> inner) : assert(inner.length == N), super(inner);
}
```

Reply:
Cool！